### ISSUE TITLE
Optimize CSS Delivery + OMGF Pro: OMGF's fonts-defining CSS file is removed

### ISSUE THREAD

**Issue by vmanthos** [wp-media/wp-rocket#4114]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅ (3.9.0.4)
- Used the search feature to ensure that the bug hasn’t been reported before ✅

**Describe the bug**

When Optimize CSS Delivery is used, alongside the [OMGF | Host Google Fonts Locally](https://wordpress.org/plugins/host-webfonts-local/) Pro plugin, their CSS file:
`wp-content/uploads/omgf/pro-merged/pro-merged.css`

which contains the `@font-face` rules to the locally hosted Google Fonts files:
https://snippi.com/s/yp3cb2f

is removed from the source code.

**To Reproduce**

1. Load some Google Fonts, and use them, e.g. for the `p` element.
2. Install OMGF | Host Google Fonts Locally and the Pro version you'll find in our [drive](https://drive.google.com/drive/folders/1zAJcAEGP1shmYz8On4tlqc5ynnF9b_-1?usp=sharing). Use the default options.
3. Check the front-end. A URL to a file like this one `wp-content/uploads/omgf/pro-merged/pro-merged.css` should be there.
4. Enable the **Optimize CSS Delivery** option, and check for the URL. It will be removed.

**Expected behavior**

The file shouldn't be removed when the OCD is active.

**Additional context**

As described in the respective [Slack thread](https://wp-media.slack.com/archives/C43T1AYMQ/p1624874551406700?thread_ts=1624855466.406400&cid=C43T1AYMQ), commenting the `async_css()` action preserves the CSS file that's created by the OMGF plugin:
https://github.com/wp-media/wp-rocket/blob/50cb2df771c028fce57c95e0d7cd1a2a1dd6bfe8/inc/Engine/CriticalPath/CriticalCSSSubscriber.php#L88

I've checked this with all WP Rocket's features disabled, including our **Optimize Google fonts.**

**Ticket:** https://secure.helpscout.net/conversation/1535931722/269808?folderId=2135277

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by KristofDR** [comment]
Hi @vmanthos 

I recently had the same(?) or at least a similar issue, where when enabling "Optimize CSS Delivery" would break font delivery by OMGF Pro.

I was in contact with the dev behind OMGF Pro, and he immediately pushed a code fix on my site which he now pushed to a new update (version 2.3.1 of the plugin, should be available) that fixed the issue for me.

Could you check again if the issue persists? For me it looks like that's fixed. (running wp rocket 3.8.8)

Cheers

**Comment by vmanthos** [comment]
Thank you for letting me know about this @KristofDR.

I confirmed this is fixed with the 2.3.1 version of OMGF Pro.

From their changelog:
```
= 2.3.1 | July 5th, 2021 =
* Fix: WP Rocket (and other CSS optimization plugins) trigger OMGF Pro multiple times. We now skip out early, if the stylesheet is already added.
 ```
